### PR TITLE
Feature/videjs error handling

### PR DESCRIPTION
**Pull request by rodrigopavezi**
Added the listener for the error event from player.
Passed the videoJs Error object to the playerError function
Created default error types and message if error object is null
Updated tests to work with new error messages and types

@donnapep @stulees please review. cheers

### CODE REVIEW

File: src/widget/player-vjs.js [hunk 1]
```diff
@@ -65,6 +65,13 @@ RiseVision.Video.PlayerVJS = function PlayerVJS( params, mode ) {
     }
   }
 
+  function _onError() {
+
+    var error = _playerInstance.error();
```
Comment (donnapep) [inline src/widget/player-vjs.js]: I don't think you need to create a variable for this since it's only used once.

File: src/widget/video.js [hunk 2]
```diff
@@ -308,12 +308,12 @@ RiseVision.Video = ( function( window, gadgets ) {
       type = "MEDIA_ERR_UNKNOWN",
       errorMessage = "Sorry, there was a problem playing the video.",
       errorTypes = [
-        "MEDIA_ERR_CUSTOM",            // = 0
```
Comment (donnapep) [inline src/widget/video.js]: I think this is probably a good example of a commit that could have been its own commit, and then merged into the previous one to create a single commit, if that makes sense.

File: test/integration/non-storage/messaging.html [hunk 3]
```diff
@@ -90,42 +90,64 @@
...
         RiseVision.Video.playerError();
 
         assert.equal(document.querySelector(".message").innerHTML, "Sorry, there was a problem playing the video.", "message text");
         assert.isTrue((document.getElementById("messageContainer").style.display === "block"), "message visibility");
       });
 
-      test("should show format error message when file could not be played", function() {
-        RiseVision.Video.playerError({ "message": "Error loading media: File could not be played" });
+      test("should show MEDIA_ERR_ABORTED error message when file could not be played", function() {
+        console.log(MediaError.errorTypes);
```
Comment (donnapep) [inline test/integration/non-storage/messaging.html]: This should be removed.

File: src/widget/video.js (hunk 2 above)
Comment (donnapep) [inline src/widget/video.js]: @rodrigopavezi I don't think you can do this at this point (at least I don't think so?). If, for example, the uncommenting had been done in its own commit, separate from the changes to the tests, then the second commit could have been squashed into the first.

My original comment was more of a discussion point and something to consider when working on future PRs.

Comment (rodrigopavezi) [inline src/widget/video.js]: @donnapep I understand, I made the uncomment change on a separated PR but when trying to squash it there was no option to squash with the first PR only with the second PR regarding the tests. I think as we can only squash with the precious PR in this case I should leave the uncomment change on a separated PR. Something to keep in mind for the future. cheers

Comment (donnapep) [inline src/widget/video.js]: @rodrigopavezi Just curious, but what command do you use to squash?

Comment (rodrigopavezi) [inline src/widget/video.js]: @donnapep sure, I usually do the following

```
$ git merge-base BRANCH master
$ git rebase -i HASH
```

Then I follow the steps to choose the commit for squashing changing p for s. It does not seem there is an option to choose which previous branch to squash. Do you know a way of doing that? cheers

Comment (rodrigopavezi) [inline src/widget/video.js]: @donnapep sorry, you are right I meant commits. In this case I had to squash A with C and keep B so we would have AC-B. I couldn't find a way to do that, I think it only allows to squash the very near commits as you showed.

Comment (donnapep) [inline src/widget/video.js]: @rodrigopavezi OK, I follow you now. Apparently you can just reorder the commits in the editor as per the "How" section of [this doc](https://www.intertech.com/Blog/git-how-to-combine-non-consecutive-multiple-commits-into-one/).

Comment (rodrigopavezi) [inline src/widget/video.js]: @donnapep awesome, I will try that next time. cheers

**Pull request merged**